mainchain-follower/mock/src/block.rs

```diff
@@ -10,17 +8,25 @@ pub struct BlockDataSourceMock {
 	mc_epoch_duration_millis: u32,
 }
 
-#[async_trait]
-impl BlockDataSource for BlockDataSourceMock {
-	async fn get_latest_block_info(&self) -> Result<MainchainBlock> {
+#[derive(Debug, Clone, PartialEq, serde::Serialize)]
+pub struct MainchainBlock {
```

Comment:
This struct is present 3 times: in db-sync-follower, mc-hash and mock. At the same time it is composed of `domain` types. What do you think about defining this struct in `domain` crate?

Reply:
Yeah, that's a good idea, at least for now. Different IDPs use different fields from this type so there's some space for reducing the APIs, but block data is still served by the same cache db-sync query, so we can re-use a common type in the APIs until their needs drift apart, if they ever do.